tests/Unit/Evolution/Systems/GeneralizedHarmonic/BoundaryConditions/Test_BjorhusImpl.cpp

```diff
@@ -0,0 +1,218 @@
+    get<1, 2, 2>(spec_bc_dt_v_zero)[i] = 90.4;
+    get<1, 2, 3>(spec_bc_dt_v_zero)[i] = 90.4;
+    get<2, 2, 2>(spec_bc_dt_v_zero)[i] = 95.6;
+    get<2, 2, 3>(spec_bc_dt_v_zero)[i] = 95.6;
+
+    get<0, 3, 3>(spec_bc_dt_v_zero)[i] = 79.;
+    get<1, 3, 3>(spec_bc_dt_v_zero)[i] = 90.4;
+    get<2, 3, 3>(spec_bc_dt_v_zero)[i] = 95.6;
+  }
+
+  // Compare values returned by BC action vs those from SpEC
+  CHECK_ITERABLE_APPROX(local_bc_dt_v_zero, spec_bc_dt_v_zero);
+
+  // Test for another set of values
+  {
+    // Setting unit_interface_normal_Vector
+    for (size_t i = 0; i < slice_grid_points; ++i) {
+      get<0>(local_unit_interface_normal_vector)[i] = -1.;
+      get<1>(local_unit_interface_normal_vector)[i] = 0.;
+      get<2>(local_unit_interface_normal_vector)[i] = 0.;
+    }
+  }
+  // Compute rhs value
+  GeneralizedHarmonic::BoundaryConditions::detail::
+      set_dt_v_zero_constraint_preserving(
+          make_not_null(&local_bc_dt_v_zero),
+          local_unit_interface_normal_vector, local_four_index_constraint,
+          local_char_projected_rhs_dt_v_zero, local_char_speeds);
+
+  // Initialize with values from SpEC
+  for (size_t i = 0; i < slice_grid_points; ++i) {
+    get<0, 0, 0>(spec_bc_dt_v_zero)[i] = 91.;
+    get<0, 0, 1>(spec_bc_dt_v_zero)[i] = 91.;
+    get<0, 0, 2>(spec_bc_dt_v_zero)[i] = 91.;
+    get<0, 0, 3>(spec_bc_dt_v_zero)[i] = 91.;
+    get<1, 0, 0>(spec_bc_dt_v_zero)[i] = 91.6;
+    get<1, 0, 1>(spec_bc_dt_v_zero)[i] = 91.6;
+    get<1, 0, 2>(spec_bc_dt_v_zero)[i] = 91.6;
+    get<1, 0, 3>(spec_bc_dt_v_zero)[i] = 91.6;
+    get<2, 0, 3>(spec_bc_dt_v_zero)[i] = 94.4;
+    get<2, 0, 0>(spec_bc_dt_v_zero)[i] = 94.4;
+    get<2, 0, 1>(spec_bc_dt_v_zero)[i] = 94.4;
+    get<2, 0, 2>(spec_bc_dt_v_zero)[i] = 94.4;
+
+    get<0, 1, 1>(spec_bc_dt_v_zero)[i] = 91.;
+    get<0, 1, 2>(spec_bc_dt_v_zero)[i] = 91.;
+    get<0, 1, 3>(spec_bc_dt_v_zero)[i] = 91.;
+    get<1, 1, 1>(spec_bc_dt_v_zero)[i] = 91.6;
+    get<1, 1, 2>(spec_bc_dt_v_zero)[i] = 91.6;
+    get<1, 1, 3>(spec_bc_dt_v_zero)[i] = 91.6;
+    get<2, 1, 1>(spec_bc_dt_v_zero)[i] = 94.4;
+    get<2, 1, 2>(spec_bc_dt_v_zero)[i] = 94.4;
+    get<2, 1, 3>(spec_bc_dt_v_zero)[i] = 94.4;
+
+    get<0, 2, 2>(spec_bc_dt_v_zero)[i] = 91.;
+    get<0, 2, 3>(spec_bc_dt_v_zero)[i] = 91.;
+    get<1, 2, 2>(spec_bc_dt_v_zero)[i] = 91.6;
+    get<1, 2, 3>(spec_bc_dt_v_zero)[i] = 91.6;
+    get<2, 2, 2>(spec_bc_dt_v_zero)[i] = 94.4;
+    get<2, 2, 3>(spec_bc_dt_v_zero)[i] = 94.4;
+
+    get<0, 3, 3>(spec_bc_dt_v_zero)[i] = 91.;
+    get<1, 3, 3>(spec_bc_dt_v_zero)[i] = 91.6;
+    get<2, 3, 3>(spec_bc_dt_v_zero)[i] = 94.4;
+  }
+
+  // Compare values returned by BC action vs those from SpEC
+  CHECK_ITERABLE_APPROX(local_bc_dt_v_zero, spec_bc_dt_v_zero);
+}
+}  // namespace
+
+SPECTRE_TEST_CASE("Unit.Evolution.Systems.GeneralizedHarmonic.BCBjorhus.VZero",
+                  "[Unit][Evolution]") {
+  // Piece-wise tests with SpEC output
+  const size_t grid_size = 3;
+
```

Comment:
I like that you're testing vs spec. Could you also just put in a random-number python test?

Reply:
added